City-of-Helsinki/hel-fi-drupal-grants
# UHF-10741: Force user to visit all pages before submitting application

# [UHF-10741](https://helsinkisolutionoffice.atlassian.net/browse/UHF-10741)
Force users to visit all form pages before submitting.

## What was done
* Check if user has visited all pages on the preview page
* Generate warning for unvisited pages
* Disable submit button if unvisited pages exist

## How to install

* Make sure your instance is up and running on correct branch.
  * `git checkout origin/UHF-10741`
  * `make fresh`
* Import translations: `drush locale:check && drush locale:update`
* Run `make drush-cr`

## How to test
* Use the search page to find the _Liikunta, laitos- ja säätiöavustukset_ application
* Select the application and start filling it. Login to the page through Suomi.fi auth if you need to.
*  Try jumping from page 1 to page 5. You should see a warning telling you that you have unvisited pages on the form.
* Try filling all the pages except page 3. **Do not visit page 3**. Afterwards navigate to page 5. You can try submitting the page by removing the `disabled` attribute from the submit button. This way you can test the `validateForm` function that should prevent the form submit if any pages are unvisited.
* Try with different forms. You should always see the warning and submit should always be disabled as long as pages remain unvisited.
* Check code quality.

## Designers review




## Automatic- / Regression tests




[UHF-10741]: https://helsinkisolutionoffice.atlassian.net/browse/UHF-10741?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
